dokka-subprojects/plugin-base/src/test/kotlin/linkableContent/LinkableContentTest.kt

```diff
@@ -418,4 +420,41 @@ class LinkableContentTest : BaseAbstractTest() {
             }
         }
     }
+
+    @Test
+    fun `should have a correct url to an external inherited member #2879`() {
+        val writerPlugin = TestOutputWriterPlugin()
+        val configuration = dokkaConfiguration {
+            sourceSets {
+                sourceSet {
+                    sourceRoots = listOf("src/")
+                    classpath = listOfNotNull(jvmStdlibPath)
+                }
+            }
+        }
+
+        testInline(
+            """
+            /src/kotlin/main.kt
+            open interface C : Collection<C>
+            interface A : C()
+            interface B : C()
+            """.trimIndent()
+            ,
+            pluginOverrides = listOf(writerPlugin),
+            configuration = configuration
+        ) {
+            renderingStage = { rootPage, ctx ->
+                val location = DokkaLocationProvider(rootPage, ctx, ".html")
+                val classA = rootPage.dfs { it is ClasslikePageNode && it.name == "A" }
+                val classB = rootPage.dfs { it is ClasslikePageNode && it.name == "B" }
+                val classC = rootPage.dfs { it is ClasslikePageNode && it.name == "C" }
+                val sourceSet = (classA as ClasslikePageNode).content.sourceSets
+                val dri = org.jetbrains.dokka.links.DRI("kotlin.collections", "Collection", Callable(name="isEmpty", receiver=null, params=emptyList()))
+                assertEquals("../-b/index.html#-719293276%2FFunctions%2F-1617659094", location.resolve(dri, sourceSet, classA))
+                assertEquals("index.html#-719293276%2FFunctions%2F-1617659094", location.resolve(dri, sourceSet, classB))
+                assertEquals("../-b/index.html#-719293276%2FFunctions%2F-1617659094", location.resolve(dri, sourceSet, classC))
```

Comment:
This is a reproducer.
If we remove an anchor logic in  `DokkaLocationProvider`, https://github.com/Kotlin/dokka/issues/3054 will be fixed as well. We have not discussed it at grooming, but it can be a big change for the user.

@IgnatBeresnev @whyoleg WDYT? 
```kt
            open interface C : List<C> {
            /**
            * [Collection.isEmpty]
            **/
            val p = 0
            }
            interface A : C()
            interface B : C()
            
```

Reply:
`p` in the example is a bit confusing, it looks like it's somehow related to the bug, but it seems like it isn't?

Given the example

```kotlin
interface C : List<C>

interface A : C
interface B : C
```

I can open the page for `B` and see inherited functions there, like `isEmpty`. If I click on the anchor link icon, it gives me a link to `B#isEmpty`, the same place where I generated it: 

```
http://localhost:63342/dokka-debug-kts/build/dokka/html/dokka-debug-kts/me.beresnev.dokka.debug/-b/index.html#-1000881820%2FFunctions%2F769193423
```

<img width="1207" alt="image" src="https://github.com/Kotlin/dokka/assets/22685910/12472c09-0059-415b-999b-42796a9690cf">

What's going to happen if we implement the change? Will the anchor link lead to `List#isEmpty` on kotlinlang.org? If yes, what's the expected behaviour for overridden functions?

```kotlin
interface C : List<C>

interface A : C
interface B : C {

    /**
     * Overriding documentation
     */
    override fun isEmpty(): Boolean = true
}
```

Reply:
> p in the example is a bit confusing, it looks like it's somehow related to the bug, but it seems like it isn't?

The reproducer is in the unit test.
```kt
interface C : List<C>
interface A : C
interface B : C
```
The `p` is not related to the bug at all, but the link `[Collection.isEmpty]` is.

> I can open the page for B and see inherited functions there, like isEmpty. If I click on the anchor link icon, it gives me a link to B#isEmpty, the same place where I generated it:

Sorry, I got it where you are confused. I mean only the `resolving anchors by DRI` logic (see `DokkaLocationProvider .resolveDrI`), but not generating anchor links for an icon (see `DokkaBaseLocationProvider.anchorForDCI`).  
 If you open the page for A or C and click on the `isEmpty ` keyword, it opens `-b/index.html#-1000881820%2FFunctions%2F769193423`.
 
 > Will the anchor link lead to List#isEmpty on kotlinlang.org?
 
 No, the generating anchor link icon will left the same.
 
> If yes, what's the expected behaviour for overridden functions?

In this case, Dokka generates a dedicated page for the overridden functions.

Reply:
Oh, I think I understand now! After the fix, all anchor icon links will work the same way as before, but if `C` inherits `isEmpty`, clicking on `C#isEmpty` will lead to kotlinlang instead of leading to itself or to A/B, right? If so, it looks logical to me 👍